src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -244,22 +244,12 @@ HRESULT CProfilerManager::AddRawProfilerHook(
 
     m_profilerCallbackHolder = std::move(profilerCallbackHolder);
 
-    if (m_attachedClrVersion != ClrVersion_2)
-    {
-        hr = pCorProfilerCallback->Initialize(m_pWrappedProfilerInfo);
-    }
-    else
-    {
-        hr = pCorProfilerCallback->Initialize(m_pRealProfilerInfo);
-    }
-
     if (FAILED(hr))
     {
         CLogging::LogError(_T("Raw profiler hook returned failure"));
         return E_FAIL;
     }
```

Comment:
I don't think that this is valid anymore. The last hr will be set to the result of the query interface, which could validly be E_NOINTERFACE. What you are doing here is making the profiler incompatible raw profilers that implement callbacks >= 7. I could see valid reasons that the raw profilers would not implement this interface (they might just not need the callbacks).

Reply:
Thanks for catching this

---
In reply to: [420238513](https://github.com/microsoft/CLRInstrumentationEngine/pull/256#discussion_r420238513) [](ancestors = 420238513)